server/src/main/java/io/druid/server/coordination/SegmentChangeRequestDrop.java

```diff
@@ -62,4 +62,29 @@ public String toString()
            "segment=" + segment +
            '}';
   }
+
+  @Override
+  public boolean equals(Object o)
+  {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+
+    SegmentChangeRequestDrop that = (SegmentChangeRequestDrop) o;
+
+    if (!segment.equals(that.segment)) {
```

Comment:
Ok, I peeked through the code and it seems DataSegmentChangeRequest seems to use more or less straight up instances of the class. 

But @nishantmonu51 : can you please add a unit test where a `DataSegmentChangeRequest` for drops and loads are read from JSON and verify that the equals method works for Jackson instanced objects?